### PR TITLE
Branches - Mira

### PR DESCRIPTION
**Pull request by calopter**
# Task List
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe in your own words what the Model is doing in Rails | The model abstracts over database transactions to make persistent data available to the controller
Describe in your own words what the Controller is doing in Rails | The controller contains the logic that integrates the model and the view.
Describe in your own words what the View is doing in Rails | The view abstracts over presenting the data operated on by the controller in html using templating
Describe an edge-case controller test you wrote | patch to an invalid id should respond with a redirect
What is the purpose of using strong params? (i.e. the params method in the controller) | 
How are Rails migrations related to Rails models? | Migrations change the schema of the database, from which ActiveRecord populates the model classes
Describe one area of Rails that are still unclear on | timezone management

### CODE REVIEW

**Comment by tildeee** [comment]
# Task List
## What We're Looking For

Feature | Feedback
:------------- | :-------------
**Baseline** |
Appropriate Git Usage with no extraneous files checked in | x
Answered comprehension questions | x
Successfully handles: Index, Show | x
Index & Show tests pass | x
Successfully handles: New, Create | x
New & Create tests pass | x
Successfully handles: Edit, Update | x
Tests for Edit & Update test valid & invalid task ids | x
Successfully handles: Destroy, Task Complete | x
Tests for Destroy & Task Complete include tests for valid and invalid task ids | x
Routes follow RESTful conventions | x
Uses named routes (like `_path`) | x
**Overall** |



Great work on this project, Mira! Your app code and your test code look great! As always, it is clear with your code that you are committed to concise code and best practices.


Lastly: I agree with your comprehension questions that the Time Zone thing is unclear-- to be quite honest, I didn't anticipate this need for such weirdness for Time Zone, so I'll let you know if I come up with anything.


Again, super great work overall, keep up the hard work!

File: test/controllers/tasks_controller_test.rb [hunk 1]
```diff
@@ -90,37 +84,95 @@
...
+      }.must_change "Task.count", -1
 
+      assert_nil Task.find_by(id: task.id)
+
+      must_respond_with :redirect
+      must_redirect_to tasks_path
+    end
+
+    it "redirects if given an invalid id" do
+      delete task_path(-1)
```
Comment (tildeee) [inline test/controllers/tasks_controller_test.rb:144]: It might be interesting to also check/test that the Task.count didn't change

File: test/controllers/tasks_controller_test.rb [hunk 2]
```diff
@@ -90,37 +84,95 @@
...
+      must_respond_with :redirect
+    end
   end
 
   # Complete for Wave 4
   describe "toggle_complete" do
-    # Your tests go here
+    it "can mark a task as completed" do
```
Comment (tildeee) [inline test/controllers/tasks_controller_test.rb:151]: Super good tests for `toggle_complete` -- keep it up!
